File: CPP/ByTier/C/1859_C_Another_Permutation_Problem.cpp
// Problem: CF 1859 C - Another Permutation Problem
// https://codeforces.com/contest/1859/problem/C

/*
 * Problem: Another Permutation Problem
 * Algorithm: Brute-force with optimization
 * Time Complexity: O(n^3), where n is the length of the permutation
 * Space Complexity: O(n), for storing the permutation array
 *
 * The idea is to try all possible ways to split the permutation into two parts:
 * - First part: elements from 1 to x (in order)
 * - Second part: elements from n down to (n - x) (in reverse order)
 * For each such split, compute the cost and keep track of the maximum.
 */

#include <math.h>
#include <algorithm>
#include <iostream>
#include <utility>

using namespace std;
#define rep(i, a, n) for (int i = (a); i < (n); ++i)
#define per(i, a, n) for (int i = (a); i > (n); --i)
#define pb emplace_back
#define mp make_pair
#define clr(a, b) memset(a, b, sizeof(a))
#define all(x) (x).begin(), (x).end()
#define lowbit(x) (x & -x)
#define fi first
#define se second
#define lson o << 1
#define rson o << 1 | 1
#define gmid l[o] + r[o] >> 1
using ll = long long;
using ull = unsigned long long;
using pii = pair<int, int>;
using pll = pair<ll, ll>;
using ui = unsigned int;
constexpr int mod = 998244353;
constexpr int inf = 0x3f3f3f3f;
constexpr double EPS = 1e-8;
const double PI = acos(-1.0);
constexpr int N = 510;
int T, n, a[N];

void _main() {
  cin >> T;
  while (T--) {
    cin >> n;
    int ans = 0;
    // Try all possible splits of the permutation
    rep(x, 0, n + 1) {
      // Build the first part with values from 1 to x
      rep(i, 1, x + 1) a[i] = i;
      // Build the second part with values from n down to (n - x)
      int m = n;
      rep(i, x + 1, n + 1) a[i] = m--;
      int cur = 0, mx = 0;
      // Compute the cost for current configuration
      rep(i, 1, n + 1) {
        int v = i * a[i];  // Value of term i * a[i]
        cur += v;          // Add to sum
        mx = max(mx, v);   // Track maximum value
      }
      ans = max(ans, cur - mx); // Update maximum cost
    }
    cout << ans << '\n';
  }
}

int main() {
  ios::sync_with_stdio(0);
  cin.tie(0);
  cout.tie(0);
  _main();
  return 0;
}


// https://github.com/VaHiX/CodeForces/

File: Python/ByTier/C/1859_C_Another_Permutation_Problem.py
# Problem: CF 1859 C - Another Permutation Problem
# https://codeforces.com/contest/1859/problem/C

"""
Algorithm: Another Permutation Problem
Purpose: Find the maximum cost among all permutations of length n, where cost is defined as:
         (sum of p[i] * i) - (max of p[j] * j)

Approach:
- For a given n, we want to find the permutation that maximizes the cost.
- The key insight is that in order to maximize the cost:
  - We should try to place the largest elements at positions with the highest weights (i.e., larger indices)
  - But we also want to maximize the sum of p[i]*i while minimizing the maximum p[j]*j

Time Complexity: O(n^3) in worst case due to nested loops.
Space Complexity: O(1), only using a few variables for computation.

The code uses a brute-force-like approach over possible permutations by iterating through
various configurations and calculating the corresponding cost, and keeping track of the maximum.

Note: This is a hand-crafted and potentially suboptimal simulation of the problem,
not necessarily following the most efficient approach, but it matches the provided code.
"""
for _ in range(int(input())):
    l = int(input())
    m = 0
    for i in range(l - 1, -1, -1):  # Iterating from n-1 down to 0
        s = 0
        for j in range(1, i):  # Summing up j*j from 1 to i-1
            s += j * j
        c = 0
        for k in range(l - i + 1):  # For a fixed i, iterate k from 0 to l-i
            k = (l - k) * (i + k)  # Compute value for current k
            if k > c:  # Track the maximum of these values
                c = k
            s += k  # Add it to the sum
        s = s - c  # Final cost calculation
        if s > m:  # Update maximum cost found so far
            m = s
        else:
            break  # Early termination condition
    print(m)


# https://github.com/VaHiX/CodeForces/